### PR TITLE
[sw/silicon_creator] Add ROM specialization for fake keys

### PR DESCRIPTION
**Pull request by alphan**
This PR
* Moves fake key definitions to a separate TU,
* Reorganizes targets and dependencies so that key definitions appear at the first level of the dependency graph as a direct dependency of ROM,
* Marks the target that defines fake RSA keys and its dependencies as `testonly`,
* Replaces getters for authorized RSA keys with link-time dependency injection, 
* Adds a ROM _specialization_ with fake keys, and
* Adds a scaffolding for real RSA key definitions.

After this PR `sigverify` code will be organized as follows:
* `//sw/device/silicon_creator/lib/sigverify`: Signature verification
* `//sw/device/silicon_creator/rom:sigverify_keys`: Authorized key lookup
* `//sw/device/silicon_creator/rom/keys/fake`: Fake keys
* `//sw/device/silicon_creator/rom/keys/real`: Real keys

Please see the individual commits for more details.

Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

**Comment by alphan** [comment]
@jadephilipoom PTAL when you get the chance. This PR should make adding a PQ signature verification algorithm implementation and the corresponding authorized keys easier. We still need to update the manifest format and `opentitantool` though.

File: sw/device/silicon_creator/rom/BUILD [hunk 1]
```diff
@@ -153,6 +153,17 @@ opentitan_rom_binary(
...
+    name = "rom_with_fake_keys",
+    testonly = True,
+    deps = [
+        ":linker_script",
+        ":rom_common",
+        ":rom_start",
+        "//sw/device/silicon_creator/rom/keys/fake",
+    ],
+)
+
```
Comment (jadephilipoom) [inline sw/device/silicon_creator/rom/BUILD:166]: Checking that I understand this change correctly: would the final ROM binary be another target that looks the same as this one except is not testonly and depends on `//sw/device/silicon_creator/rom/keys/real`?

**Review by jadephilipoom — APPROVED** [review]
This looks like a nice refactor! :slightly_smiling_face: It's especially elegant that the fake keys are marked testonly so we don't accidentally use them for a non-test target.

File: sw/device/silicon_creator/rom/BUILD (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/rom/BUILD:166]: Correct, that's the plan. Since we haven't generated any _real_ keys, I didn't define that target yet.

**Comment by alphan** [comment]
Merging now for bake time. @moidx @cfrantz I'll check in with you next week to see if you have any comments.

**Pull request merged**